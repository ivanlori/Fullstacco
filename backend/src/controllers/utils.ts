import { NextFunction } from "express"

interface Error {
	statusCode: number;
}

export const handleErrorStatus = (
	err: Error,
	next: NextFunction | null
) => {
	if (err) {
		err.statusCode = 500
	}

	if (next) {
		next(err)
	}
}