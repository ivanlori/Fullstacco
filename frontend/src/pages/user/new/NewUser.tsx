import { ReactElement } from 'react'

import { useIntl } from 'react-intl'

import { UserForm } from '../_components/UserForm'

export const NewUser = (): ReactElement => {
	const { formatMessage } = useIntl()

	return (
		<div className="lg:pl-24">
			<div className="xl:mx-auto flex">
				<div className="flex flex-col w-full">
					<div className="bg-white text-gray_900 rounded-xl p-5 m-10">
						<UserForm
							user={null}
							title={formatMessage({
								id: 'user.create.update.title.create'
							})} />
					</div>
				</div>
			</div>
		</div>
	)
}
