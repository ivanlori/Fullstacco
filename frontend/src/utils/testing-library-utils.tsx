import { PropsWithChildren } from "react"

import { PreloadedState } from "@reduxjs/toolkit"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import { createIntl, RawIntlProvider } from "react-intl"
import { Provider } from "react-redux"

import { AppStore, RootState, setupStore } from "store"

import enMsg from '../languages/en.json'

const intl = createIntl({
	locale: 'en',
	messages: enMsg
})

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
	preloadedState?: PreloadedState<RootState>
	store?: AppStore
	withIntl?: boolean
}

export function renderWithProviders(
	ui: React.ReactElement,
	{
		preloadedState = {},
		// Automatically create a store instance if no store was passed in
		store = setupStore(preloadedState),
		withIntl,
		...renderOptions
	}: ExtendedRenderOptions = {}
) {
	function Wrapper({ children }: PropsWithChildren<unknown>): JSX.Element {
		return withIntl ? (
			<Provider store={store}>
				<RawIntlProvider value={intl}>
					{children}
				</RawIntlProvider>
			</Provider>
		) : <Provider store={store}>{children}</Provider>
	}
	return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

export const renderWithIntlProvider = (
	component: React.ReactElement,
	{ ...options } = {},
): RenderResult => {
	const wrapper = () => (
		<RawIntlProvider value={intl}>
			{component}
		</RawIntlProvider>
	)
	return render(component, { wrapper, ...options })
}
