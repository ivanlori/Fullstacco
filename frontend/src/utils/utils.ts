import { AxiosError, AxiosResponse } from "axios"

import { IUserState } from "types/user"

export const getUserId = () => localStorage.getItem('userId')

export const isAuthenticated = () => getToken() && getUserId()

export const getToken = () => localStorage.getItem('tk')

export const isAdmin = (user: IUserState) => user.role === 0

export const isAccount = (user: IUserState) => user.role === 1

export const handleError = (err: unknown): AxiosResponse<unknown, unknown> => {
	const error = err as AxiosError

	return error.response as AxiosResponse<unknown, unknown>
}
