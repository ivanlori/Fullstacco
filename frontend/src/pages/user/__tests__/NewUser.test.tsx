import '@testing-library/jest-dom'
import { MemoryRouter } from 'react-router-dom'

import { renderWithProviders } from 'utils/testing-library-utils'

import NewUser from '../new/NewUser'

describe('NewUser', () => {

	test('should match snapshot', () => {
		const { container } = renderWithProviders(
			<MemoryRouter>
				<NewUser />
			</MemoryRouter>,
			{
				withIntl: true
			}
		)

		expect(container).toMatchSnapshot()
	})
})
