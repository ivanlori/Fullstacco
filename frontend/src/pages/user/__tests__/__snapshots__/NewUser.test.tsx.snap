// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewUser should match snapshot 1`] = `
<div>
  <div
    class="lg:pl-24"
  >
    <div
      class="xl:mx-auto flex"
    >
      <div
        class="flex flex-col w-full"
      >
        <div
          class="bg-white text-gray_900 rounded-xl p-5 m-10"
        >
          <div
            class="p-10"
          >
            <div
              class="flex justify-between mb-5"
            >
              <h1
                class="text-2xl font-medium"
              >
                Create User
              </h1>
            </div>
            <form
              class="text-gray_900"
            >
              <div
                class="mb-4"
              >
                <label>
                  Name
                </label>
                <div
                  class="relative"
                >
                  <input
                    class="Input"
                    data-testid="user-form-name"
                    name="name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="mb-4"
              >
                <label>
                  Lastname
                </label>
                <div
                  class="relative"
                >
                  <input
                    class="Input"
                    data-testid="user-form-lastname"
                    name="lastname"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="mb-4"
              >
                <label>
                  Username
                </label>
                <div
                  class="relative"
                >
                  <input
                    class="Input"
                    data-testid="user-form-username"
                    name="username"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="mb-4"
              >
                <label>
                  Email
                </label>
                <div
                  class="relative"
                >
                  <input
                    class="Input"
                    data-testid="user-form-email"
                    name="email"
                    type="email"
                    value=""
                  />
                </div>
              </div>
              <div
                class="mb-4"
              >
                <label>
                  Roles
                </label>
                <div
                  class=" css-b62m3t-container"
                >
                  <span
                    class="css-1f43avz-a11yText-A11yText"
                    id="react-select-2-live-region"
                  />
                  <span
                    aria-atomic="false"
                    aria-live="polite"
                    aria-relevant="additions text"
                    class="css-1f43avz-a11yText-A11yText"
                  />
                  <div
                    class=" css-1s2u09g-control"
                  >
                    <div
                      class=" css-319lph-ValueContainer"
                    >
                      <div
                        class=" css-qc6sy-singleValue"
                      >
                        Admin
                      </div>
                      <div
                        class=" css-6j8wv5-Input"
                        data-value=""
                      >
                        <input
                          aria-autocomplete="list"
                          aria-expanded="false"
                          aria-haspopup="true"
                          autocapitalize="none"
                          autocomplete="off"
                          autocorrect="off"
                          class=""
                          id="react-select-2-input"
                          role="combobox"
                          spellcheck="false"
                          style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                          tabindex="0"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class=" css-1hb7zxy-IndicatorsContainer"
                    >
                      <span
                        class=" css-1okebmr-indicatorSeparator"
                      />
                      <div
                        aria-hidden="true"
                        class=" css-tlfecz-indicatorContainer"
                      >
                        <svg
                          aria-hidden="true"
                          class="css-tj5bde-Svg"
                          focusable="false"
                          height="20"
                          viewBox="0 0 20 20"
                          width="20"
                        >
                          <path
                            d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                          />
                        </svg>
                      </div>
                    </div>
                  </div>
                  <input
                    name="role"
                    type="hidden"
                    value="0"
                  />
                </div>
              </div>
              <div
                class="mb-4"
              >
                <label>
                  Password
                </label>
                <div
                  class="relative"
                >
                  <input
                    class="Input"
                    data-testid="user-form-password"
                    name="password"
                    type="password"
                    value=""
                  />
                </div>
              </div>
              <div
                class="flex justify-end"
              >
                <button
                  class="Btn BtnPrimary"
                  data-testid="user-form-save"
                  type="submit"
                >
                  Save
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
